src/main/java/core/basesyntax/data/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.data.dao;
+
+import core.basesyntax.data.model.FruitTransaction;
+import core.basesyntax.data.model.OperationType;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    private static final String INPUT_DATA = "fruit_shop_transactions.csv";
+
+    @Override
+    public List<FruitTransaction> readFile(String fileName) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new java.io.FileReader(fileName))) {
+            String line;
+            br.readLine();
+            while ((line = br.readLine()) != null) {
+                String[] data = line.split(",");
```

Comment:
Consider adding a check to ensure that `data` contains the expected number of elements (at least 3) before accessing them. This will prevent `ArrayIndexOutOfBoundsException` if the CSV file has malformed lines.